### ISSUE TITLE
Null_resource should export triggers attributes

### ISSUE THREAD

**Issue by arr-dev** [hashicorp/terraform#6154]
### Terraform Version

0.6.14
### Affected Resource(s)

Please list the resources as a list, for example:
- null_resource
### Terraform Configuration Files

``` hcl
resource "digitalocean_droplet" "do-node" {
  count = "${var.node_count}"
  image = "${var.node_image}"
  name = "${var.node_name}${format("%02d", count.index + 1)}"
  region = "${var.node_region}"
  size = "${var.node_size}"
  private_networking = true
  ssh_keys = ["${split(",", var.ssh_keys)}"]
  lifecycle {
    prevent_destroy = true
  }
}

resource "template_file" "do-node" {
  count = "${var.node_count}"
  template = "${path.module}/node.json.tpl"
  vars {
    ipaddress = "${element(digitalocean_droplet.do-node.*.ipv4_address, count.index)}"
    private_ipaddress = "${element(digitalocean_droplet.do-node.*.ipv4_address_private, count.index)}"
  }
}

resource "null_resource" "do-node" {
  count = "${var.node_count}"
  triggers {
    json = "${element(template_file.do-node.*.rendered, count.index)}"
    node = "${element(digitalocean_droplet.do-node.*.name, count.index)}"
  }

  provisioner "local-exec" {
    command = <<EOF
    [ -f ${path.cwd}/nodes/${element(digitalocean_droplet.do-node.*.name, count.index)}.json ] && exit 0
    cat <<JSON > ${element(digitalocean_droplet.do-node.*.name, count.index)}.json
${element(template_file.do-node.*.rendered, count.index)}
JSON
EOF
  }
}
```
### Expected Behavior

Attributes in `triggers` map should be exported in `null_resource`, 
so they can be accessible as e.g. `${self.node}` or `${self.triggers.node}`  in provisioner, to continue on above example.

``` hcl
resource "null_resource" "do-node" {
  count = "${var.node_count}"
  triggers {
    json = "${element(template_file.do-node.*.rendered, count.index)}"
    node = "${element(digitalocean_droplet.do-node.*.name, count.index)}"
  }

  provisioner "local-exec" {
    command = <<EOF
    [ -f ${path.cwd}/nodes/${self.node}.json ] && exit 0
    cat <<JSON > ${self.node}.json
${self.json}
JSON
EOF
  }
}
```
### Actual Behavior

Actual behavior requires a lot of repetition and possibility for errors.

**Comment by apparentlymart** [comment]
Hi @arr-dev! Sorry for the super-delayed response here.

Although at times this is not super clear in the documentation, in general *any* attribute of a resource can be interpolated, even if it's not listed under the heading about exported attributes. So in fact what you're describing here should be possible, and I do some very similar things in configs at my workplace... the triggers are, as you've seen, a convenient place to "stash" some values to use within the provisioners.

Here's your example with the syntax adjusted for what I think should work:

```hcl
esource "null_resource" "do-node" {
  count = "${var.node_count}"
  triggers {
    json = "${element(template_file.do-node.*.rendered, count.index)}"
    node = "${element(digitalocean_droplet.do-node.*.name, count.index)}"
  }

  provisioner "local-exec" {
    command = <<EOF
    [ -f ${path.cwd}/nodes/${self.triggers["node"]}.json ] && exit 0
    cat <<JSON > ${self.triggers["node"]}.json
${self.triggers["json"]}
JSON
EOF
  }
}
```

Although you originally asked about Terraform 0.6.14 because this is such an old issue, the syntax I showed above is actually for Terraform 0.7 after the map indexing operator was added. If you're still on the 0.6 line then the syntax would instead look like `self.triggers.node`, because 0.6 syntax didn't distinguish between attributes and map keys.

I'm going to cllose this because I think what you want to do here is already possible, but do let me know if I'm mistaken and then we can either re-open this or file a new issue to capture a more specific detail of the problem here.
